docs/using-ngrok-with/k8s.md

```diff
@@ -0,0 +1,246 @@
+    :::
+
+    ```yaml showLineNumbers
+    apiVersion: v1
+    kind: Service
+    metadata:
+    name: game-2048
+    spec:
+    ports:
+        - name: http
+        port: 80
+        targetPort: 80
+    selector:
+        app: game-2048
+    ---
+    apiVersion: apps/v1
+    kind: Deployment
+    metadata:
+    name: game-2048
+    spec:
+    replicas: 1
+    selector:
+        matchLabels:
+        app: game-2048
+    template:
+        metadata:
+        labels:
+            app: game-2048
+        spec:
+        containers:
+            - name: backend
+            image: alexwhen/docker-2048
+            ports:
+                - name: http
+                containerPort: 80
+    ---
+    # ngrok Ingress Controller Configuration
+    apiVersion: networking.k8s.io/v1
+    kind: Ingress
+    metadata:
+      name: game-2048-ingress
+    spec:
+      ingressClassName: ngrok
+      rules:
+        - host: my-awesome-k8s-cluster.ngrok.app
+          http:
+          paths:
+            - path: /
+            pathType: Prefix
+            backend:
+              service:
+              name: game-2048
+              port:
+                number: 80
+    ```
+
+1. Apply the manifest file:
+    
+    ```bash
+    kubectl apply -f ngrok-manifest.yaml
+    ```
+
+1. To confirm the manifest is successfully applied, go to the [ngrok Dashboard](https://dashboard.ngrok.com) and click [Edge Configurations](https://dashboard.ngrok.com/edge-configurations). You should see a new Edge Configuration for your cluster with the name matching your URL (1) — for example: `my-awesome-k8s-cluster.ngrok.app`. Also note that your some of your cluster configurations are presented int the dashboard as annotations (2).
+
+    ![ingress created](/img/howto/ingress-controller/k8s-ingress-app-1.png)
+
+1. Access your ingress URL (i.e. `my-awesome-k8s-cluster.ngrok.app`) to confirm the 2048 app is accessible from the internet:
+    
+    ![application public](/img/howto/ingress-controller/k8s-ingress-app-2.png)
+
+## Step 3: Add edge security to your app
+
+The ngrok Ingress Controler for Kubernetes provides custom resource definitions (CRDs) for additional edge features available in ngrok. In this example, we're expanding the Ingress Controller with Google OAuth to allow access only from users with the email domains `@acme.com` or `@ngrok.com` and to apply a circuit breaker to your app at 80%. These features are enforced at the ngrok edge, ensuring only authorized users can access your app.
+
+:::tip Notes:
+This example is very similar to the previous version, with the following changes:
+- Lines 40-42: Associates your Ingress Controller with the configuration `oauth-and-circuit-breaking` via annotation
+- Lines 56-72: Sets the edge configuration as a custom CRD (NgrokModuleSet).
+- Lines 62-67: Sets the circuit breaker module over 50% threshold.
+- Lines 68-72: Sets the OAuth module to allow access only for users with the email domains `@acme.com` or `@ngrok.com`.
```

Comment:
hmmm, these would probably be easier to follow and keep up to date if we could just show a diff between the files with some callout, but i don't know of a great tool for that tbh. The service and deployment aren't changed, maybe cutting down that noise would help here. 

Something like this maybe shows more clearly the diff but makes it harder to copy paste the right code

![image](https://user-images.githubusercontent.com/8029578/229940702-594b000f-e8c4-4b1c-83ab-040543031963.png)

Reply:
Yeah. Docusaurus doesn't provide diff ootb. 

The closest I can get is a line highlighter like in go.
https://ngrok-docs-git-frederico-ngrok-ingress-controller-ngrok-dev.vercel.app/docs/using-ngrok-with/go/

Reply:
The new version uses highlight. Unfortunately we cannot have just the diff without compromising the copy/pasta.

Reply:
oooh a built in way with docusaurus one day would be cool. also just a future thought